tests/test_block_factory.py

```diff
@@ -27,13 +27,13 @@
 VOCAB_SIZE = 64
 
 
-@pytest.mark.parametrize("attn_dropout", [0.0, 0.1])
-@pytest.mark.parametrize("residual_dropout", [0.0, 0.1])
+@pytest.mark.parametrize("attn_dropout", [0.1])
```

Comment:
Is this just to reduce the amount of testing?

Reply:
ah yes sorry I should have added a note. At some point the tests were just too long, it was more about a triton version that I wanted to bump up (and where the initial JIT is slower) but I thought that maybe it that it was a good idea to reduce that test set (especially since I added deepnorm). We have other tests where dropout 0 and >0 is tested so the code coverage did not change, that's the metric I was checking to see whether this made sense. 

I remember that this was discussed some time ago with Cole I think, no super strict take personally (dropout 0 and >0 are different codepaths), I'm kind of changing my mind on the topic (as long as the code coverage is maintained). Is that ok for you ?

Reply:
Sounds good! Since we're testing both codepaths elsewhere seems fine to me :)